### PR TITLE
add eslint and precommit hooks

### PR DESCRIPTION
**Pull request by rana123bilal**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,14 +17,27 @@
...
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
```
Comment (Vladlan) [inline package.json:35]: Please configure list-staged and husky properly - it is not working.

Comment (Vladlan) [reply package.json:35]: https://typicode.github.io/husky/#/

File: package.json [hunk 2]
```diff
@@ -17,14 +17,27 @@
     "react-dom": "^18.2.0",
     "react-scripts": "5.0.1",
     "styled-components": "^5.3.6",
-    "web-vitals": "^2.1.4"
+    "web-vitals": "^2.1.4",
+    "husky": "^8.0.1",
+    "lint-staged": "^13.0.3"
```
Comment (Vladlan) [inline package.json:22]: where is eslint?

Comment (Vladlan) [reply package.json:22]: When you will install eslint fix all its errors and warnings

**Review by Vladlan — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (rana123bilal) [reply package.json:37]: there was an issue in the code in configuration of husky, now it is working perfectly.

File: package.json (hunk 2 above)
Comment (rana123bilal) [reply package.json]: eslint is now installed and all fixed all errors.

**Review by Vladlan — CHANGES_REQUESTED** [review]

**Comment by Vladlan** [comment]
@rana123bilal please merge this branch to main

**Pull request closed without merging**